Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift

```diff
@@ -34,8 +34,12 @@ import Foundation
         self.mySegments.removeAllObjects()
     }
     
-    public func getSegments() -> [String] {
-        return self.mySegments.allKeys as! [String]
+    public func getSegments() -> [String]? {
+        return self.mySegments.allKeys as? [String]
+    }
+    
+    public func getSegments(key: String) -> [String]? {
```

Comment:
can you check this?

Reply:
check if we can remove this call that does not use key at all and behaves the same as getSegments with no key

Reply:
This is related to the previous comment about InMemoryMySegmentsCache$.addSegments methods.